examples/va/nfv/sriov/edpm/values.yaml

```diff
@@ -1,89 +1,19 @@
-#     i.e.
-#     - type: sriov_pf
-#       name: eno4      <---- change
-# - Change "spec.nodeTemplate.ansible.ansibleVars.edpm_sshd_allowed_ranges[0]" to
-#   match the CIDR of your control plane network, if necessary
-# - Change "edpm_bootstrap_command" to include subscription-manager and podman login
-#   commands to register the EDPM nodes as in https://access.redhat.com/solutions/253273
-# - Change the "spec.nodes[].ansible.ansibleHost" and "spec.nodes[].networks[0].fixedIP" values
-#   to match the control plane IPs of your nodes, if necessary
-# - Change "spec.nodeTemplate.ansible.ansibleVars.edpm_kernel_args" to be the desired kernel
-#   args for your environment
-# - Change "spec.nodeTemplate.ansible.ansibleVars.edpm_tuned_isolated_cores" to be the desired
-#   CPU core isolation for your environment
-# - Change "spec.nodeTemplate.ansible.ansibleVars.edpm_neutron_sriov_agent_SRIOV_NIC_physical_device_mappings"
-#   to the desired mapping of a network name to your SRIOV interface
-#
-
-apiVersion: dataplane.openstack.org/v1beta1
-kind: OpenStackDataPlaneNodeSet
+# local-config: referenced, but not emitted by kustomize
+apiVersion: v1
+kind: ConfigMap
 metadata:
-  name: openstack-edpm-sriov
-spec:
-  env:
-    - name: ANSIBLE_FORCE_COLOR
-      value: "True"
-  services:
-    - bootstrap
-    - configure-network
-    - validate-network
-    - install-os
-    - configure-os
-    - run-os
-    - reboot-os
-    - libvirt
-    - ovn
-    - neutron-sriov
-    - neutron-metadata
-    - nova-custom-sriov
-  preProvisioned: true
-  nodes:
-    edpm-compute-0:
-      ansible:
-        ansibleHost: 192.168.122.100 # CHANGEME
-      hostName: edpm-compute-0
-      networks:
-      - defaultRoute: true
-        fixedIP: 192.168.122.100 # CHANGEME
-        name: CtlPlane
-        subnetName: subnet1
-      - name: InternalApi
-        subnetName: subnet1
-      - name: Storage
-        subnetName: subnet1
-      - name: Tenant
-        subnetName: subnet1
-    edpm-compute-1:
-      ansible:
-        ansibleHost: 192.168.122.101 # CHANGEME
-      hostName: edpm-compute-1
-      networks:
-      - defaultRoute: true
-        fixedIP: 192.168.122.101 # CHANGEME
-        name: CtlPlane
-        subnetName: subnet1
-      - name: InternalApi
-        subnetName: subnet1
-      - name: Storage
-        subnetName: subnet1
-      - name: Tenant
-        subnetName: subnet1
-  networkAttachments:
-    - ctlplane
-  nodeTemplate:
-    ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
-    managementNetwork: ctlplane
+  name: edpm-values
+  annotations:
+    config.kubernetes.io/local-config: "true"
+data:
+  ssh_keys:
+    # Authorized keys that will have access to the dataplane computes via SSH
+    authorized: CHANGEME
+    # The private key that will have access to the dataplane computes via SSH
+    private: CHANGEME2
+    # The public key that will have access to the dataplane computes via SSH
+    public: CHANGEME3
```

Comment:
by the way one thing to be aware of is the secretGenerator can use files directly

so if we just want to put these directly into a secreate we can do that with kustomize.

```
secretGenerator:
- name: app-tls
  files:
  - secret/tls.cert
  - secret/tls.key
  type: "kubernetes.io/tls"
```


https://kubectl.docs.kubernetes.io/references/kustomize/builtins/#_secretgenerator_

Reply:
yea I was looking at that approach, along with https://polarsquad.com/blog/loading-dynamic-configurations-in-kubernetes-kustomize which might save a few remote objects needing to be created and help with a more GitOps approach.

I'm going to experiment with both either next week while I'm in Westford, or when I return and see which approach seems to make more sense from a productization/documentation perspective.

Reply:
well given our intent is to store these sshkeys in a secrete use the secretGenerator would be the recommended approach or more idiomatic approach to take.

you can use env files

https://github.com/SeanMooney/nova-operator/tree/zuul-jobs-part-2/ci/nova-operator-base/topology/base

or config maps in general to make either entire fields or individual fields available in the customize context but if you intend to store files in a secret or configmap at the end of the customization then our default choice should be the ConfigmapGenerator or SecretGenerator that are part fo customize.

if we need something else then we can reach for more advanced concepts

using the env file approach vules.yamls would have to be converted to env format but it would look like this

```
configMapGenerator:
- name: environment-variables
  envs: [vaules.env]
  behavior: create

replacements:
- source:
    fieldPath: data.OPENSTACK_OPERATOR_INDEX_URL
    kind: ConfigMap
    name: environment-variables
    version: v1
  targets:
  - fieldPaths:
    - spec.image
    select:
      group: operators.coreos.com
      kind: CatalogSource
      name: openstack-operator-index
      namespace: openstack
      version: v1alpha1
```

where the env file has 
```
OPENSTACK_OPERATOR_INDEX_URL=quay.io/openstack-k8s-operators/openstack-operator-index:latest
```

anyway we can come back to this in a follow up pr